react/ProductCustomizer.js

```diff
@@ -1,54 +1,180 @@
+    const selectedVariations = this.state.selectedVariations
+
+    const updatedSelectedArray = selectedVariations.filter((currentVariation) => {
+      return variationToRemove === undefined || (variationToRemove.Id !== currentVariation.Id)
+    })
+
+    this.setState({ selectedVariations: updatedSelectedArray })
+  }
+
+  /**
+  * onHandleAddSelectedVariations
+  * Add a new selected variation to array then updates total amount.
+  * @param object variation
+  * @return void
+  */
+  onHandleAddAndCalculateSelectedVariations = (variation) => {
+    const selectedVariations = this.state.selectedVariations
+
+    selectedVariations.push(variation)
+    this.setState({ selectedVariations: selectedVariations })
+
+    this.calculateTotal()
+  }
+
+  /**
+  * calculateTotal
+  * Calculate total amount of selected variations
+  * @return void
+  */
+  calculateTotal = () => {
+    let currentTotal = 0
+    const selectedVariations = this.state.selectedVariations
+
+    if (selectedVariations.length === 0) {
+      return 0
+    }
+
+    for (const selectedVariation of selectedVariations) {
+      currentTotal += selectedVariation.price
+    }
+
+    this.setState({ total: currentTotal })
+  }
+
+  /**
+  * handleOpenModal
+  * Show the product modal
+  * @return void
+  */
+  handleOpenModal = () => {
     this.setState({ isModalOpen: true })
   }
 
-  handleCloseModal() {
+  /**
+  * handleCloseModal
+  * Close the product modal
+  * @return void
+  */
+  handleCloseModal = () => {
     this.setState({ isModalOpen: false })
   }
 
+  handleSubmit = e => {
+    console.log(this.state.selectedVariations)
+    e.preventDefault()
+
+    // Checl Lists validations
+    // Add product to cart
+  }
+
   render() {
+    const {
+      variations,
+      selectionsValidated,
+      onHandleUpdateAmount,
+    } = this.state
+
+    const { enableChangeToppings } = this.props
+
     return (
       <div>
         <Button onClick={this.handleOpenModal}>Customize your product</Button>
 
         <Modal
-          top
-          isMobile="true"
+          centered
           isOpen={this.state.isModalOpen}
           onClose={this.handleCloseModal}
         >
           <div className={'flex-ns h-100-ns'}>
-            <div className={'w-100 w-third-ns bg-light-gray bb b--light-gray flex-ns items-center-ns  h-100-ns'}>
-              <h1 className={`vtex-product-customizer__title tc f4 fw5 absolute ma0 pa5 w-100 bg-black-40 white ${!isMobile ? 'dn' : ''}`}>Create Your Own</h1>
+            <h1 className={`vtex-product-customizer__title tc f4 fw5 ma0 pa5 w-100 bg-black-40 white ${!isMobile ? 'dn' : ''}`}>Create Your Own</h1>
+            <div className={'w-100 w-third-ns flex-ns tc items-center-ns pa5 h-100-ns'}>
               <img
-                className={'vtex-product-customizer__image'}
-                alt="Create Your Own"
-                src="https://via.placeholder.com/670x500"
+                className={'vtex-product-customizer__image br3'}
```

Comment:
Here too. :pick:

Reply:
Fixed in commit 2d2092c522f572a871eea9c57d10a73b4837ee5f